app/src/test/java/org/mozilla/focus/utils/DownloadUtilsTest.java

```diff
@@ -0,0 +1,62 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.utils;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mozilla.focus.web.Download;
+import org.robolectric.RobolectricTestRunner;
+
+import static junit.framework.Assert.assertEquals;
+
+@RunWith(RobolectricTestRunner.class)
+public class DownloadUtilsTest {
+
+    private static void assertContentDisposition(String expected, String contentDisposition) {
+        Download download = new Download(null, null, contentDisposition, null, 0, null);
+        assertEquals(expected, DownloadUtils.guessFileName(download));
+    }
+
+    @Test
+    public void testGuessFileName_contentDisposition() {
+        // Default file name
+        assertContentDisposition("downloadfile.bin", "");
+        assertContentDisposition("downloadfile.bin", "attachment");
+        assertContentDisposition("downloadfile.bin", "attachment;");
+        assertContentDisposition("downloadfile.bin", "attachment; filename\"filename.jpg\"; foo");
+
+        // Provided filename field
+        assertContentDisposition("filename.jpg", "attachment; filename=\"filename.jpg\"");
```

Comment:
\[PMD\] INFO: The String literal "filename\.jpg" appears 4 times in this file; the first occurrence is on line 32